src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts

```diff
@@ -118,11 +125,12 @@ export class AdminHomePageComponent {
   }
 
   /**
-   * Adds all the pending and failed-to-add instructors.
+   * Adds all the pending and failed-to-add instructors one by one.
+   * Refer to https://github.com/TEAMMATES/teammates/issues/11039
    */
-  addAllInstructors(): void {
+  async addAllInstructors(): Promise<void> {
```

Comment:
I think what you are looking for could be the `concat` function. It subscribes to observables sequentially. The current method definitely feels like a bit of a hack

https://rxjs-dev.firebaseapp.com/api/index/function/concat

Reply:
Yes. I did try `concat` offline, but the tricky part is that: the error return cannot be traced back to the instructor who failed to be created. Also, the `JoinLink` return has no info of instructor and I have to use a pipe to map it to an indexed tuple for use too.

Reply:
Any idea how the error handling should go since in concat, errors are treated anonymously?

Reply:
You could declare a new type like `InstructorErrorRequest` then encapsulate error message + index/instructor data in the error. Then `catchError` in the `pipe` and `throwError` the new type with the data of the instructor and error message. Then you could use the data in the `concat` call to react accordingly. Just a possible idea. If it doesn't really work out, then the current method is fine too.

Reply:
Ah, I was trying with the `catchError` part but not the `throwError` part. Let me see.